flink-runtime/src/test/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointTest.java

```diff
@@ -22,50 +22,279 @@
+import org.apache.flink.runtime.resourcemanager.ArbitraryWorkerResourceSpecFactory;
+import org.apache.flink.runtime.resourcemanager.JobLeaderIdService;
+import org.apache.flink.runtime.resourcemanager.ResourceManager;
+import org.apache.flink.runtime.resourcemanager.ResourceManagerFactory;
+import org.apache.flink.runtime.resourcemanager.ResourceManagerRuntimeServices;
+import org.apache.flink.runtime.resourcemanager.ResourceManagerRuntimeServicesConfiguration;
+import org.apache.flink.runtime.resourcemanager.StandaloneResourceManagerFactory;
+import org.apache.flink.runtime.resourcemanager.TestingJobLeaderIdService;
+import org.apache.flink.runtime.resourcemanager.TestingResourceManager;
+import org.apache.flink.runtime.resourcemanager.slotmanager.SlotManager;
+import org.apache.flink.runtime.resourcemanager.slotmanager.SlotManagerBuilder;
+import org.apache.flink.runtime.rpc.FatalErrorHandler;
+import org.apache.flink.runtime.rpc.RpcService;
+import org.apache.flink.runtime.util.ExecutorThreadFactory;
+import org.apache.flink.runtime.util.SignalHandler;
+import org.apache.flink.util.ConfigurationException;
 
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
+import javax.annotation.Nullable;
+
 import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
 
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
 import static org.junit.Assert.fail;
 
 /** Tests for the {@link ClusterEntrypoint}. */
 public class ClusterEntrypointTest {
 
+    private static final long TIMEOUT_MS = 3000;
+
+    private Configuration flinkConfig;
+    private ExecutorService testingExecutor;
+
+    @Before
+    public void before() {
+        flinkConfig = new Configuration();
+        testingExecutor =
+                Executors.newSingleThreadExecutor(new ExecutorThreadFactory("testing-executor"));
+    }
+
+    @After
+    public void tearDown() {
+        testingExecutor.shutdownNow();
+    }
+
     @Test(expected = IllegalConfigurationException.class)
     public void testStandaloneSessionClusterEntrypointDeniedInReactiveMode() {
-        Configuration configuration = new Configuration();
-        configuration.set(JobManagerOptions.SCHEDULER_MODE, SchedulerExecutionMode.REACTIVE);
-        new MockEntryPoint(configuration);
+        flinkConfig.set(JobManagerOptions.SCHEDULER_MODE, SchedulerExecutionMode.REACTIVE);
+        new TestingEntryPoint.Builder().setConfiguration(flinkConfig).build();
         fail("Entrypoint initialization is supposed to fail");
     }
 
-    private static class MockEntryPoint extends ClusterEntrypoint {
+    @Test
+    public void testCloseAsyncShouldNotCleanUpHAData() throws Exception {
```

Comment:
Should we also add a test which ensures that the HA data is cleaned up and we deregister the application if the `DispatcherResourceManagerComponent` completes the `shutdownFuture` successfully?